package/sbp_nmea2000_bridge/src/sbp_nmea2000_bridge.cpp

```diff
@@ -0,0 +1,917 @@
+                                     void *context) {
+      UNUSED(sender_id);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n"
+        << "\tGot " << static_cast<u16>(len) << "/"
+        << sizeof(tracking_channel_state_t) << "="
+        << static_cast<u16>(len / sizeof(tracking_channel_state_t))
+        << " states. Message len is most likely statically allocated.\n";
+
+      constexpr u8 cConstellationMaxSats = 32;
+      std::array<std::pair<u8, u8>, cConstellationMaxSats> sats_gps = {};
+      std::array<std::pair<u8, u8>, cConstellationMaxSats> sats_glo = {};
+      auto sats_gps_end_it = sats_gps.begin();
+      auto sats_glo_end_it = sats_glo.begin();
+      auto sbp_tracking_state = reinterpret_cast<msg_tracking_state_t*>(msg);
+      for(u8 i = 0; i < (len / sizeof(tracking_channel_state_t)); ++i) {
+        auto tracking_channel_state = sbp_tracking_state->states[i];
+        if(tracking_channel_state.sid.sat == 0) {  // Invalid sat.
+          continue;
+        }
+
+        switch(tracking_channel_state.sid.code) {
+          case 0:  // GPS satellites. Fallthrough.
+          case 1:
+          case 2:
+          case 5:
+          case 6:
+            *sats_gps_end_it = std::make_pair(tracking_channel_state.sid.sat,
+                                              tracking_channel_state.cn0);
+            ++sats_gps_end_it;
+            break;
+          case 3:  // GLONASS satellites. Fallthrough.
+          case 4:
+            *sats_glo_end_it = std::make_pair(tracking_channel_state.sid.sat,
+                                              tracking_channel_state.cn0);
+            ++sats_glo_end_it;
+            break;
+          default:
+            d << "\tControl should have never reached this point.\n";
+            exit(-1);  // Should never happen.
+        }
+      }
+
+      // Sort and remove duplicates.
+      auto comparator_lower_than = [](const std::pair<u8, u8>& a,
+                                      const std::pair<u8, u8>& b) {
+          return a.first < b.first;
+      };
+      auto comparator_equal = [](const std::pair<u8, u8>& a,
+                                 const std::pair<u8, u8>& b) {
+          return a.first == b.first;
+      };
+      std::sort(sats_gps.begin(), sats_gps_end_it, comparator_lower_than);
+      std::sort(sats_glo.begin(), sats_glo_end_it, comparator_lower_than);
+      sats_gps_end_it = std::unique(sats_gps.begin(), sats_gps_end_it,
+                                    comparator_equal);
+      sats_glo_end_it = std::unique(sats_glo.begin(), sats_glo_end_it,
+                                    comparator_equal);
+
+      // Triton2 screen that this was tested with acted weird in regards
+      // to sat count:
+      //  - up to 12 sats (inclusive), everything was displayed fine
+      //  - from 13 to 17 sats (inclusive), number 12 would be shown on screen
+      //  - from 18 and upwards, no sat count number would be shown
```

Comment:
Can we just use a hash table to begin with, instead of de-duplicating at the end?